server/swap/swap.go

```diff
@@ -1221,48 +1240,39 @@ func (s *Swapper) checkInaction(assetID uint32) {
 
 		switch match.Status {
 		case order.NewlyMatched:
-			if match.makerStatus.swapAsset != assetID {
-				return
-			}
-			// If the maker is not acting, the swapTime won't be set. Check against
-			// the time the match notification was sent (match.time) for the broadcast
-			// timeout.
-			if match.makerStatus.swapTime.IsZero() && match.time.Before(oldestAllowed) {
+			// Maker has not broadcast their swap. They have until match time
+			// plus bTimeout.
+			if match.time.Before(oldestAllowed) {
 				failMatch(true)
 			}
 		case order.MakerSwapCast:
-			if match.takerStatus.swapAsset != assetID {
-				return
-			}
```

Comment:
These swapAsset checks were not necessary, but in this MakerSwapCast case, it should have been checking makerStatus.swapAsset not takerStatus.swapAsset.